### PR TITLE
Introduce and use vmlinux crate

### PR DESCRIPTION
**Pull request by danielocfb**
Please see individual commits for descriptions.

### CODE REVIEW

File: vmlinux/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,161 @@
+#![allow(non_upper_case_globals)]
+
+use std::env::consts::ARCH;
+use std::path::Path;
+use std::path::PathBuf;
+
+// TODO: Ideally we'd deduplicate contents at this level (as well; not just on
```
Comment (anakryiko) [inline vmlinux/src/lib.rs:7]: those are also huge text files which would compress very nicely, but I assume it's impossible in const context?

**Review by anakryiko — APPROVED** [review]
LGTM. Do you plan to convert libbpf-bootstrap rust examples to this as well?

File: vmlinux/src/lib.rs (hunk 1 above)
Comment (d-e-s-o) [reply vmlinux/src/lib.rs:7]: Likely not in a `const` context, yeah. We could probably have a function based interface if need be and load and cache stuff on demand. I mostly got nerd-sniped here and wanted to make it work as I had originally envisioned...

The other avenue would to exclude unnecessary headers based on features. Either way, it's probably not urgent unless/until we use this functionality in anything but example & test code.

**Comment by d-e-s-o** [comment]
> LGTM. Do you plan to convert libbpf-bootstrap rust examples to this as well?

Not immediately, but if we decide to publish this crate eventually then yes, happy to adjust the code there.

**Pull request merged**